core/src/main/scala/org/apache/spark/TestUtils.scala

```diff
@@ -364,6 +369,22 @@ private[spark] object TestUtils {
     }
   }
 
+  def withHttpServer(resBaseDir: String = ".")(body: URL => Unit): Unit = {
```

Comment:
Shall we move this to `org.apache.spark.deploy.k8s.integrationtest.Utils` since this aims for `integration tests` and we don't use it in other module? Last time, did you have another use case?

Reply:
I think this a very useful structure and we could use it even for testing Spark submits better.
So I would like to leave this here, if it is OK for you.